src/jcpsantiago/bulgogi.clj

```diff
@@ -31,7 +55,27 @@
   "
   [req -ns]
   (let [{:keys [input-data features]} req
-        fns (resolved-features features -ns)
+        fns (memoized-features features -ns)
+        coeffects (memoized-coeffects fns)
         fn-ks (map keyword features)]
-    (->> (transformed input-data fns)
+    (->> (transformed (enriched input-data coeffects) fns)
          (zipmap fn-ks))))
+
+(defn all-features 
+  "Returns a map of feature-name -> feature-var"
+  []  
+  (->> (all-ns)
+       (filter #(::features (meta %)))
+       (map ns-publics)
+       (apply merge-with #(throw (Exception. (str "Conflict between: " %1 " and :" %2))))))
+
+(defn- feature-conflicts? []
+  (->> (all-features)       
+       (mapcat keys)
+       distinct?))
+
+(all-features)
```

Comment:
was this just for testing, or is there another behind calling it here?

Reply:
This is my current try for compile time collision checking. Requiring the ns should throw a compile time error if you have a collision